src/utils/hooks/useSetting.ts

```diff
@@ -0,0 +1,23 @@
+import {useCallback} from 'react';
+import {useDispatch} from 'react-redux';
+
+import {getParsedSettingValue, setSettingValue} from '../../store/reducers/settings';
+
+import {useTypedSelector} from './useTypedSelector';
+
+export const useSetting = <T>(key: string, defaultValue: T): [T, (value: T) => void] => {
+    const dispatch = useDispatch();
+
+    const settingValue: T = useTypedSelector(
+        (state) => getParsedSettingValue(state, key) || defaultValue,
+    );
+
+    const setValue = useCallback(
```

Comment:
why did you decide to use `useCallback` here?

Reply:
I decided to add `useCallback` to ensure, that return values are properly memoized and could be safely used as dependencies in other hooks